### DISCUSSION TITLE
encodevertexbuffer after encodefilter in JS

### DISCUSSION THREAD

**Discussion by meijuji202** [zeux/meshoptimizer#358] [Q&A] [answered]
encodevertexbuffer after encodefilter in JS

		var data = new Float32Array([
		    1, 0, 0, 0,
		    0, -1, 0, 0,
		    0.7071068, 0, 0.707168, 1,
		    -0.7071068, 0, -0.707168, 1,
		]);

		var encoded = encoder.encodeFilterOct(data, 4, 8, 12);
		var vertexBuffer= encoder.encodeVertexBuffer(
			encoded,
			data.length,
			4
		);

**report errors**

**Comment by zeux** [comment, marked as answer]
*please* actually specify exact errors for questions like this. This report is not useful on its own and I have to spend time trying to figure out what could have possibly went wrong. One common mistake is that you need to wait for `encoder.ready` promise before actually using the encoder. Wrapping your example in `encoder.ready.then(() => { ... })` makes it work.
